lib/json-schema/schema/reader.rb

```diff
@@ -98,6 +98,8 @@ def read_uri(uri)
         else
           raise JSON::Schema::ReadRefused.new(uri.to_s, :uri)
         end
+      rescue OpenURI::HTTPError => error
```

Comment:
I'd prefer if we raised a custom error class here, eg. `JSON::Schema::ReadFailed` rather than re-raising the error of a third-party library (in case we ever changed what we use to load external files)